File: API/DTOs/OrderDto.cs
using System;
using System.Collections.Generic;


namespace API.DTOs
{
    public class OrderDto
    {
        public int Id { get; set; }
        public DateTime OrderDate { get; set; }
        public DateTime RequiredDate { get; set; }
        public DateTime? ShippedDate { get; set; }
        public DateTime RequiredReturnDate { get; set; }
        public DateTime? ReturnDate { get; set; }
        public string Status { get; set; }
        public string Comments { get; set; }
        public UserDto Deliveryman { get; set; }
        public UserDto Customer { get; set; }
        public AddressDto ShippedAddress { get; set; }
        public AddressDto ReturnAddress { get; set; }
        public ICollection<RealProductDto> OrderProducts { get; set; }
        public int ReturnAddressId { get; set; }
        public int ShippedAddressId { get; set; }
        public int CustomerId { get; set; }
        public int DeliverymanId { get; set; }
    }
}

File: client/src/app/_models/orderDto.ts
import { Address } from "./address";
import { RealProduct } from "./realProduct";
import { User } from "./user";

export interface OrderDto {
    id: number;
    orderDate: Date;
    requiredDate: Date;
    shippedDate?: Date;
    requiredReturnDate: Date;
    returnDate?: Date;
    status: string;
    comments: string;
    deliveryman: User;
    deliverymanReturn: User;
    customer: User;
    shippedAddress: Address;
    returnAddress: Address;
    orderProducts: RealProduct[];
    returnAddressId: number;
    shippedAddressId: number;
    customerId: number;
    deliverymanId: number;
    deliverymanReturnId: number;
    totalPrice: number;
}